docs/standard/io/file-path-formats.md

```diff
@@ -0,0 +1,170 @@
+- They are fully qualified DOS paths; that is, they begin with a drive letter, a volume separator, and a component separator (`C:\`).
+- They designate a legacy device (`CON`, `LPT1`).
+- They are relative to the root of the current drive; that is, they begin with a single component separator (`\`).
+- They are relative to the current directory of a specified drive; that is, they begin with a drive letter, a volume separator, and no component separator (`C:`).
+- They are relative to the current directory; that is, they begin with anything else (`temp\testfile.txt`).
+
+The type of the path determines whether or not a current directory is applied in some way. It also determines what the "root" of the path is.
+
+### Handling legacy devices
+
+If the path is a legacy DOS device such as `CON`, `COM1`, or `LPT1`, it is converted into a device path by prepending `\\.\` and returned.
+
+### Applying the current directory
+
+If a path isn't fully qualified, Windows applies the current directory to it. UNCs and device paths do not have the current directory applied. Neither does a full drive with separator C:\.
+
+If the path starts with a single component separator, the drive from the current directory is applied. For example, if the file path is `\utilities` and the current directory is `C:\temp\`, normalization produces `C:\utilities`.
+
+If the path starts with a drive letter, volume separator, and no component separator, the last current directory set from the command shell for the specified drive is applied. If the last current directory was not set, the drive alone is applied. For example, if the file path is `D:sources`, the current directory is `C:\Documents\`, and the last current directory on drive D: was `D:\sources\`, the result is `D:\sources\sources`. These "drive relative" paths are a common source of program and script logic errors. Assuming that a path beginning with a letter and a colon isn't relative is obviously not correct.
+
+If the path starts with something other than a separator, the current drive and current directory are applied. For example, if the path is `filecompare` and the current directory is `C:\utilities\`, the result is `C:\utilities\filecompare\`.
+
+> [!IMPORTANT]
+> Relative paths are dangerous in multithreaded applications (that is, most applications) because the current directory is a per-process setting. Any thread can change the current directory at any time. 
+
+### Canonicalizing separators
+
+All forward slashes (`/`) are converted into the standard Windows separator, the back slash (`\`). If they are present, series of slashes that follow the first two slashes are collapsed into a single slash.
+
+> [!IMPORTANT]
+> It is only when identifying paths for normalization purposes that the direction of the slash does not matter. Forward slashes are not supported in Windows outside of this normalization step. This is critically important if normalization is skipped.
```

Comment:
While this is technically true, most (if not all) Win32 APIs normalize so you only see this if you call NT APIs directly. Apps and command-line utilities often don't support forward slashes as they only look for back slashes and/or expect forward slashes to be option switches.